crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space/metric_exchange.rs

```diff
@@ -2,13 +2,42 @@ use super::*;
 use crate::wire::MetricExchangeMsg;
 use kitsune_p2p_types::config::KitsuneP2pTuningParams;
 use kitsune_p2p_types::dht_arc::DhtArcSet;
+use tokio::time::Instant;
+
+struct ShouldTrigger {
+    last_sync: Option<Instant>,
+    freq: u128,
+}
+
+impl ShouldTrigger {
+    pub fn new(freq: u128) -> Self {
+        Self {
+            last_sync: None,
+            freq,
+        }
+    }
+
+    pub fn should_trigger(&mut self) -> bool {
+        let now = Instant::now();
+        if self
+            .last_sync
+            .map(|s| now.saturating_duration_since(s).as_millis() > self.freq)
```

Comment:
I'm always interested in removing the ambiguity of millis vs micros, so I would prefer using types to disambiguate:

```suggestion
struct ShouldTrigger {
    last_sync: Option<Instant>,
    freq: Duration,
}

impl ShouldTrigger {
    pub fn new(freq: Duration) -> Self {
        Self {
            last_sync: None,
            freq,
        }
    }

    pub fn should_trigger(&mut self) -> bool {
        let now = Instant::now();
        if self
            .last_sync
            .map(|s| now.saturating_duration_since(s) > self.freq)
```